packages/cf-util-http-poll/src/httpPoll.js

```diff
@@ -48,12 +48,14 @@ export default function httpPoll(url, opts) {
   function onSuccess(res) {
     currentAbort = null;
     if (isCancelled) return;
-    if (!opts.isComplete(res)) {
-      intervalId = setTimeout(next, interval);
-    } else {
-      cleanup();
-      opts.onComplete(res);
-    }
+    opts.isComplete(res).then(completed => {
```

Comment:
No need, `fetch` always return a resolved promise. The only exception is when you have a syntax or type error in the params to `fetch()`, which is unrecoverable and caught in `cf-util-http`.